blog/human-readable-txs-learnings/index.mdx

```diff
@@ -0,0 +1,275 @@
+  </div>
+</div>
+
+I started this article with blind-signing but the problem is actually even larger: it's transaction safety. It means nothing if you can read a transaction but it does not do what you read or if you're talking to the wrong contract. It's layers and layers and layers of security. Which also means layers and layers and layers of coordination effort. That's why it's so difficult.
+
+
+:::note
+By no means this is a complete guide. I'm more than happy to extend this. Please get in touch if you have anything to add!
+:::
+
+# Transaction Safety
+
+## 1. Source Code Verification
+
+This is a MUST. Period. Without verification there's no way to tell what a contract really does. 
+
+DO NOT INTERACT WITH CONTRACTS THAT ARE UNVERIFIED
+
+The contract should be verified by a verifier, but ideally it is verified at multiple places. Even better if the verification is a ["perfect/exact match"](/docs/full-vs-partial-match/), this gives you a cryptographic guarantee that what you see is exactly what the contract author deployed, byte-by-byte.
+
+Players here: [Blockscout](https://www.blockscout.com), [Etherscan](https://etherscan.io/), [Routescan](https://routescan.io/), [Tenderly](https://tenderly.co/), Sourcify, and the [Verified Alliance](https://verifieralliance.org/) initiative.
+
+Wallets MUST warn users about unverified contracts and use multiple sources to check verification. Unfortunately, even in 2025, not many do even this simple requirement. 
+
+The tricky thing here is to not lead the user to believe verified == safe. A verified contract can still be malicious and verification does not check what the contract does.
+
+## 2. Labeling
+
+The second step is to make sure you are talking to the correct contract. You think you are making a swap on Uniswap but is it really Uniswap or just a verified Uniswap copy at another address? Maybe the contract you are talking to is already marked malicious and it'd be such a shame to lose money to an already known scammer. 
+
+So this entails whitelists and blacklists of contracts. I don't have deep knowledge around this but I don't really recall seeing contract labels in my wallets. There are "web3 security" companies or Etherscan labels but the problem with these is everyone solves it for themselves. 
+
+I'll be saying this many times: 
+
+**In order to solve this problem once and for all, THE SOLUTION HAS TO BE OPEN.** 
+
+Closed, siloed data will only get us so far. 
+
+Here the project I want to highlight is the [Open Labels Initiative](https://openlabelsinitiative.org/) stewarded by [growthepie.xyz](https://www.growthepie.xyz/). If there are other solutions/projects worth sharing, please share them.
+
+## 3. Audits
```

Comment:
I don't know if this is the right place, but I think it's worth mentioning https://www.securityalliance.org/ in the topic of human readable tx security.

(This is a good link to share: https://www.youtube.com/watch?v=q1JVGxgj7xM)

Reply:
SEAL does not directly fit in this article, I see it more as an incidence response group as of now.